packages/core/src/data-module/data-cache/caching/caching.spec.ts

```diff
@@ -369,6 +370,142 @@ describe('getDateRangesToRequest', () => {
+        end: END_DATE,
+        cacheSettings: DEFAULT_CACHE_SETTINGS,
+      })
+    ).toEqual([]);
+  });
+  it('converts date ranges to request informations, setting fetchMostRecentBeforeEnd only on the latest request', () => {
+    const START_DATE = new Date(2000, 0, 0);
+    const END_DATE = new Date(2001, 0, 0);
+    const CACHE_START_DATE = new Date(2000, 3, 0);
+    const CACHE_END_DATE = new Date(2000, 4, 0);
+    expect(
+      getRequestInformations({
+        request: {
+          viewport: { start: START_DATE, end: END_DATE },
+          settings: { fetchFromStartToEnd: true, fetchMostRecentBeforeEnd: true },
+        },
+        store: {
+          [STREAM_ID]: {
+            0: {
+              id: STREAM_ID,
+              resolution: 0,
+              requestHistory: [],
+              isLoading: false,
+              isRefreshing: false,
+              dataCache: EMPTY_CACHE,
+              requestCache: createDataPointCache({
+                start: CACHE_START_DATE,
+                end: CACHE_END_DATE,
+              }),
+            },
+          },
+        },
+        resolution: '0',
+        dataStreamId: STREAM_ID,
+        start: START_DATE,
+        end: END_DATE,
+        cacheSettings: DEFAULT_CACHE_SETTINGS,
+      })
+    ).toEqual([
+      expect.objectContaining({
+        start: START_DATE,
+        end: CACHE_START_DATE,
+        id: STREAM_ID,
+        resolution: '0',
+        fetchMostRecentBeforeEnd: false,
+        fetchFromStartToEnd: true,
+      }),
+      expect.objectContaining({
+        start: CACHE_END_DATE,
+        end: END_DATE,
+        id: STREAM_ID,
+        resolution: '0',
+        fetchMostRecentBeforeEnd: true,
+        fetchFromStartToEnd: true,
+      }),
+    ]);
+  });
+  it('appends fetchMostRecentBeforeStart request information', () => {
+    const START_DATE = new Date(2000, 0, 0);
+    const END_DATE = new Date(2001, 0, 0);
+    const CACHE_START_DATE = new Date(2000, 3, 0);
+    expect(
+      getRequestInformations({
+        request: {
+          viewport: { start: START_DATE, end: END_DATE },
+          settings: { fetchMostRecentBeforeStart: true },
+        },
+        store: {
+          [STREAM_ID]: {
+            0: {
+              id: STREAM_ID,
+              resolution: 0,
+              requestHistory: [],
+              isLoading: false,
+              isRefreshing: false,
+              dataCache: EMPTY_CACHE,
+              requestCache: createDataPointCache({
+                start: CACHE_START_DATE,
+                end: END_DATE,
+              }),
+            },
+          },
+        },
+        resolution: '0',
+        dataStreamId: STREAM_ID,
+        start: START_DATE,
+        end: END_DATE,
+        cacheSettings: DEFAULT_CACHE_SETTINGS,
+      })
+    ).toEqual([
+      expect.objectContaining({
+        start: new Date(0, 0, 0),
+        end: START_DATE,
+        id: STREAM_ID,
+        resolution: '0',
+        fetchMostRecentBeforeStart: true,
```

Comment:
I'm confused, why would we want to fetch the most recent before year 0?

Reply:
Yeah this approach builds the correct slice to request prior to passing it to the source, but that's actually a different paradigm than what is done for `beforeEnd` so i'll change it to pass the viewport in and have the source request the leading point.